### ISSUE TITLE
 The prediction results between djl and pytorch are different on the same model

### ISSUE THREAD

**Issue by mengpengfei** [deepjavalibrary/djl#1116]
The inference result of djl is different from The inference result of pytorch。I had check all my code，but I still can not find the problem,please help me.blow is the code of pytorch and the code of djl

![image](https://user-images.githubusercontent.com/13211758/126422963-d82a9dd6-9825-4511-a974-114a826b117b.png)
![image](https://user-images.githubusercontent.com/13211758/126424033-d80e79bf-b82d-42a9-a320-83dc354b6204.png)

**Comment by frankfliu** [comment]
@mengpengfei Please see: https://github.com/deepjavalibrary/djl/issues/970

DJL resize use PyTorch [interpolate C++ API](https://pytorch.org/docs/master/generated/torch.nn.functional.interpolate.html). The implementation is slightly different from torchvision's resize.

**Comment by mengpengfei** [comment]
I use opencv  to resize the picture,then the problem is solved.
Can you make djl compatible with OpenCV？

**Comment by frankfliu** [comment]
@mengpengfei 
It's hard to choose which implementation to use, there are many different implementations:
1. Java ImageIO
2. OpenCV
2. PIL
3. PyTorch interpolate C++ API
4. MXNet (different flavor of OpenCV)

None of above produce the exactly the same result. In ideal case, the resize algorithm should match the model was trained on. For DJL, choosing OpenCV mean break other models.

We made a few assumption:
1. Assume user will use native engine's resize algorithm (MXNet user use MXNet resize to train the model)
2. PyTorch is working on moving python based resize into there C++ API, once this done, python and C++ API will have the same behavior (There is an issue on pytorch github)
3. In most cases, the different won't impact the final inference result (the accuracy impact is small)
